src/ServerRequestFactory.php

```diff
@@ -469,6 +469,7 @@ private static function createUploadedFileFromSpec(array $value)
      */
     private static function normalizeNestedFileSpec(array $files = [])
     {
+        $normalizedFiles = [];
```

Comment:
The `UploadedFile` objects were being appended to the `$files` array, which meant that the `tmp_name` and other attributes preceeded them in the array. When same array gets passed to the validation function, it was trying to then validate strings such as the `tmp_name` as an instance of `UploadedFile`, and failing. So I just created a new array to store the `UploadedFile` objects in and returned that instead.